cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggeringInfoHelper.java

```diff
@@ -0,0 +1,197 @@
+          (DefaultProgramStatusTriggerInfo) triggerInfos.get(0);
+        return new TriggeringInfo.ProgramStatusTriggeringInfo(scheduleId,
+                                                              programStatusTriggerInfo.getRuntimeArguments(),
+                                                              getProgramRunId(programStatusTriggerInfo));
+      case TIME:
+        DefaultTimeTriggerInfo timeTriggerInfo = (DefaultTimeTriggerInfo) triggerInfos.get(0);
+        return new TriggeringInfo.TimeTriggeringInfo(scheduleId, new HashMap<>(), timeTriggerInfo.getCronExpression());
+      case OR:
+        return new TriggeringInfo.OrTriggeringInfo(getTriggeringInfoList(triggerInfos, scheduleId),
+                                                   scheduleId, null,
+                                                   triggeringPropertyMapping(triggeringScheduleInfo.getProperties()));
+      case AND:
+        return new TriggeringInfo.AndTriggeringInfo(getTriggeringInfoList(triggerInfos, scheduleId),
+                                                    scheduleId, null,
+                                                    triggeringPropertyMapping(triggeringScheduleInfo.getProperties()));
+      case PARTITION:
+        DefaultPartitionTriggerInfo partitionTriggerInfo =
+          (DefaultPartitionTriggerInfo) triggerInfos.get(0);
+        return new TriggeringInfo.PartitionTriggeringInfo(scheduleId, null,
+                                                          partitionTriggerInfo.getDatasetName(),
+                                                          partitionTriggerInfo.getDatasetNamespace(),
+                                                          partitionTriggerInfo.getExpectedNumPartitions(),
+                                                          partitionTriggerInfo.getActualNumPartitions());
+      default:
+        return null;
+    }
+  }
+
+  private static ProgramRunId getProgramRunId(DefaultProgramStatusTriggerInfo programStatusTriggerInfo) {
+    return new ProgramRunId(programStatusTriggerInfo.getNamespace(), programStatusTriggerInfo.getApplicationName(),
+                            programTypeMappings.get(programStatusTriggerInfo.getProgramType()),
+                            programStatusTriggerInfo.getProgram(), programStatusTriggerInfo.getRunId().getId());
+  }
+
+  private static TriggeringPropertyMapping triggeringPropertyMapping(Map<String, String> properties) {
+    if (!properties.containsKey(TRIGGERING_PROPERTIES_MAPPING)) {
+      return null;
+    }
+    return gson.fromJson(properties.get(TRIGGERING_PROPERTIES_MAPPING), TriggeringPropertyMapping.class);
+  }
+
+  /**
+   * ProgramType mappings for {@link ProgramType} and {@link io.cdap.cdap.api.app.ProgramType}
+   */
+  static Map<io.cdap.cdap.api.app.ProgramType, ProgramType> programTypeMappings = new HashMap<>();
+
+  static {
+    programTypeMappings.put(io.cdap.cdap.api.app.ProgramType.WORKFLOW, ProgramType.WORKFLOW);
```

Comment:
The proto ProgramType has a `getApiProgramType` method. You should use that instead of doing it manually.